torch/onnx/_internal/fx/_pass.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+import abc
+import difflib
+
+from typing import Any, Callable, Dict, Generator, Sequence, Tuple
+
+import torch
+import torch.fx
+from torch.onnx._internal import _beartype
+from torch.onnx._internal.fx import diagnostics
+
+
+class FxReadableGraphDiffer(difflib.Differ):
+    """Differ specialized for fx readable graph.
+
+    The only difference between this and official `difflib.Differ` is the `autojunk`
+    argument for `difflib.SequenceMatcher` class. In this class, `autojunk` is set
+    to `False`. This is to prevent `difflib.SequenceMatcher` recognizing stacktrace
+    messages in fx readable graph as junk, as these messages tend to be long (>200)
+    and repeat multiple times, which falls under the junk filter criteria.
+
+    `Reference: Automatic junk heuristic <https://docs.python.org/3/library/difflib.html>`_
+    """
+
+    def compare(self, a: Sequence[str], b: Sequence[str]) -> Generator[str, None, None]:
```

Comment:
I can see all the type annotations, but I still don't know what does this function do. Can you provide some example inputs and example outputs?

Reply:
docstring is up there which mentioned the difference between this and the official Differ class, and why we need it.

Please let me know if you see there is a better way, I don't like having to "copy" all the code here neither.

Reply:
For now I'm adding a docstring for compare to mention this again. I'm open to other suggestions.

Reply:
I tried another approach with context manager, please take a look. And I didn't add docstring for compare, that way it inherits parent's docstring. Added comment for context manager.